src/Tui.hs

```diff
@@ -48,7 +48,34 @@ getUrlFromGitRepository = do
   let url = head . tail . words $ head $ lines remote
   case "github" `isInfixOf` url of
     True -> return $ AutoUrl $ formGithubUrl url
-    False -> return Fail
+    False ->
+      case "gitlab" `isInfixOf` url of
+        True -> return $ AutoUrl $ formGitlabUrl url
+        False -> return Fail
+
+gitlabIssuesUrlPrefix = "https://gitlab.com/api/v4/projects/"
+
+-- /commento%2Fcommento/issues?state=opened
+formGitlabUrl :: String -> String
+formGitlabUrl ('g':'i':'t':rest) =
+  gitlabIssuesUrlPrefix ++
+  (replaceBackSlash . giturl) rest ++ "/issues?state=opened"
+formGitlabUrl ('h':'t':'t':'p':rest) =
+  gitlabIssuesUrlPrefix ++
+  (replaceBackSlash . httpurl) rest ++ "/issues?state=opened"
+
+replaceBackSlash :: String -> String
+replaceBackSlash ('/':xs) = "%2F" ++ xs
+replaceBackSlash (x:xs) = x : replaceBackSlash xs
+
+-- assumes url is of form 'git@github.com:user/repo.git' or 'git@gitlab.com:user/repo.git`
+giturl :: String -> String
+giturl (':':xs) = removeGitSuffix xs
+giturl (x:xs) = giturl xs
+
+-- assumes url is of form 'https://github.com/user/repo.git
+httpurl ('m':'/':xs) = removeGitSuffix xs
+httpurl (x:xs) = httpurl xs
```

Comment:
Since there's an assumption being made, you should use type to keep your programs safe by returning a `Maybe String` instead.

Reply:
I'll keep this in mind while improving the code. I've added a separate issue for this: https://github.com/jnduli/hidd/issues/4